optimum/habana/diffusers/pipelines/auto_pipeline.py

```diff
@@ -0,0 +1,624 @@
+# coding=utf-8
+# Copyright 2024 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Adapted from: https://github.com/huggingface/diffusers/blob/v0.26.3/src/diffusers/pipelines/auto_pipeline.py
+- Added GAUDI_PREFIX_NAME to support Gaudi pipeline.
+- Only AutoPipelineForText2Image and AutoPipelineForInpainting are retained. No modifications were made to them.
+"""
+
+from collections import OrderedDict
+
+from diffusers.configuration_utils import ConfigMixin
+from huggingface_hub.utils import validate_hf_hub_args
+
+from .stable_diffusion import (
+    GaudiStableDiffusionInpaintPipeline,
+    GaudiStableDiffusionPipeline,
+)
+from .stable_diffusion_xl import (
+    GaudiStableDiffusionXLInpaintPipeline,
+    GaudiStableDiffusionXLPipeline,
+)
+
+
+GAUDI_PREFIX_NAME = "Gaudi"
+
+AUTO_TEXT2IMAGE_PIPELINES_MAPPING = OrderedDict(
+    [
+        ("stable-diffusion", GaudiStableDiffusionPipeline),
+        ("stable-diffusion-xl", GaudiStableDiffusionXLPipeline),
+    ]
+)
+
+
+AUTO_INPAINT_PIPELINES_MAPPING = OrderedDict(
+    [
+        ("stable-diffusion", GaudiStableDiffusionInpaintPipeline),
+        ("stable-diffusion-xl", GaudiStableDiffusionXLInpaintPipeline),
+    ]
+)
+
+
+SUPPORTED_TASKS_MAPPINGS = [
+    AUTO_TEXT2IMAGE_PIPELINES_MAPPING,
+    AUTO_INPAINT_PIPELINES_MAPPING,
+]
+
+
+def _get_task_class(mapping, pipeline_class_name, throw_error_if_not_exist: bool = True):
+    def get_model(pipeline_class_name):
+        for task_mapping in SUPPORTED_TASKS_MAPPINGS:
+            for model_name, pipeline in task_mapping.items():
+                if pipeline.__name__ == pipeline_class_name:
+                    return model_name
+
+    pipeline_class_name = GAUDI_PREFIX_NAME + pipeline_class_name
+    model_name = get_model(pipeline_class_name)
+
+    if model_name is not None:
+        task_class = mapping.get(model_name, None)
+        if task_class is not None:
+            return task_class
+
+    if throw_error_if_not_exist:
+        raise ValueError(f"AutoPipeline can't find a pipeline linked to {pipeline_class_name} for {model_name}")
+
+
+class AutoPipelineForText2Image(ConfigMixin):
```

Comment:
Why cant we inherit this from diffusers this class and only override HPU specific methods?

Reply:
Done.